Update packaging instructions (#513)
openSUSE/agama: PACKAGING.md

OBS](https://build.opensuse.org/project/show/YaST:Head:Agama). This document summarizes the
process we follow to build those packages.

The process to build each package is slightly different depending on the technology we are using.
While the Ruby-based one (`rubygem-agama`) is built as any other YaST package, the web UI
(`cockpit-agama`) and the CLI (`agama-cli`) rely on [OBS source
services](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.source_service.html).

## Versioning policy

We have decided to follow a single number schema: 1, 2, 3, etc. However, if we need to release a
hot-fix, we might use a dotted version (e.g., 2.1). Moreover, all the components share the same
version number. Releasing a new version implies that all of them get the new number, no matter if
they contain changes or not.

## Bumping the version

In order to release a new version, we need to:

1. Update the version number in the `service/VERSION` file with the new number. These file is read
  when building the `rubygem-agama` package.

> 2. Run `bundle install` from the `service` directory.

Reviewer: ```suggestion
2. `(cd service; bundle install)` # Updates Gemfile.lock which is part of the repository
```

> 3. Add entries in the changes files.

Reviewer: ```suggestion
3. Add entries in the changes files.
    `osc vc service/package`
    `osc vc rust/package`
    `osc vc web/package`
```

4. Open a pull request to get these changes into the repository.
5. Once the pull request is merged, tag the repository with the proper version number. The processes
   to build `cockpit-agama` and `agama-cli` use this information to infer the version. You can set
   the tag with something like:

>       git tag --sign 1 --message "Version 1"

Reviewer: JFYI:

This looks a lot like [the instructions I have for ruby-dbus](https://github.com/mvidner/ruby-dbus/blob/c7677b911e207405e50e73fa3e93170bbf29fe8d/HOWTO-RELEASE), and I've found that I want to automate it as much as possible, which for me means writing lines that  I can paste into the terminal. Example:

```
Edit VERSION NEWS.md package/rubygem-ruby-dbus.spec

    osc vc package
    git commit -m "Releasing version $(cat VERSION)."
    rake check:changelog # implies :package
    gem2rpm --template opensuse --config package/gem2rpm.yml \
      package/ruby-dbus-$(cat VERSION).gem > package/rubygem-ruby-dbus.spec
    git commit -a --amend
    rake osc:build

Push the versioned commit; tag

    cat VERSION && git tag v$(cat VERSION)
    git push
    git push --tags
```

>       git tag --sign 1 --message "Version 1"

Reviewer: ```suggestion
      git tag --sign $(cat service/VERSION) --message "Version $(cat service/VERSION)"
```

I also see that we're tagging with a bare number. Even systemd which uses ordinal versioning pefixes the tags with a `v`, like `v256`. Just saying :)

Author: Good point. Actually, I think that the service is smart enough to handle the `v` prefix.

Author: I tried and it seems to work. So I have updated the instructions accordingly.
